crates/libs/core/src/types/client/health.rs

```diff
@@ -0,0 +1,138 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+use mssf_com::FabricTypes::{
+    FABRIC_HEALTH_REPORT, FABRIC_HEALTH_REPORT_KIND_APPLICATION, FABRIC_HEALTH_REPORT_KIND_CLUSTER,
+    FABRIC_HEALTH_REPORT_KIND_DEPLOYED_APPLICATION,
+    FABRIC_HEALTH_REPORT_KIND_DEPLOYED_SERVICE_PACKAGE, FABRIC_HEALTH_REPORT_KIND_INVALID,
+    FABRIC_HEALTH_REPORT_KIND_NODE, FABRIC_HEALTH_REPORT_KIND_PARTITION,
+    FABRIC_HEALTH_REPORT_KIND_SERVICE, FABRIC_HEALTH_REPORT_KIND_STATEFUL_SERVICE_REPLICA,
+    FABRIC_HEALTH_REPORT_KIND_STATELESS_SERVICE_INSTANCE,
+};
+
+/// Wrapper of FABRIC_HEALTH_REPORT
+pub enum HealthReport {
+    Invalid,
+    StatefulServiceReplica(StatefulServiceReplicaHealthReport),
+    StatelessServiceInstance(StatelessServiceInstanceHealthReport),
+    Partition(PartitionHealthReport),
+    Node(NodeHealthReport),
+    Service(ServiceHealthReport),
+    Application(ApplicationHealthReport),
+    DeployedApplication(DeployedApplicationHealthReport),
+    DeployedServicePackage(DeployedServicePackageHealthReport),
+    Cluster(ClusterHealthReport),
+}
+
+// TODO: Implement the conversion functions for the HealthReport enum properly.
```

Comment:
Would be good. Also seems like it might be worth a macro or something to avoid having to maintain the bidirectional match statements by hand. But then again, macros can impede clarity and this doesn't change often.

Reply:
Done